drivers/spi/spi-atmel.c

```diff
@@ -1272,11 +1278,11 @@ static int atmel_spi_setup(struct spi_device *spi)
 	asd->csr = csr;
 
 	dev_dbg(&spi->dev,
-		"setup: bpw %u mode 0x%x -> csr%d %08x\n",
-		bits, spi->mode, spi->chip_select, csr);
+		"setup: bpw %u mode 0x%x -> cs%d csr%d %08x\n",
+		bits, spi->mode, spi->chip_select, spi_chip_select_capped, csr);
 
 	if (!atmel_spi_is_v2(as))
-		spi_writel(as, CSR0 + 4 * spi->chip_select, csr);
+		spi_writel(as, CSR0 + 4 * spi_chip_select_capped, csr);
 
```

Comment:
I do feel that there is a pending  issue here. It may be related to the use of internal CS selection. But it's not related to cs-gpio... So we have to find another way to prevent this crash.